scripts/delete_old_version_testing_branches.sh

```diff
@@ -3,9 +3,35 @@
 set -eo pipefail
 
 today=$(date +%Y%m%d)
-echo "today: $today"
-git fetch --prune
-version_testing_branches=$(git branch -r | grep -E '(version-testing-[0-9]{8})' | grep -v "$today" | awk -F'/' '{print $2}')
-echo "branches to be removed:"
-echo "$version_testing_branches"
-echo "$version_testing_branches" | xargs -I {} git push origin --delete {}
+echo "Today: $today"
+if [ "$CI" = true ]; then
+    unshallow_flag="--unshallow"
```

Comment:
Would be nice to add a comment explaining this.
1. What does `--unshallow` do?
2. Why do we need it only when running in CI?

Reply:
1. [the doc](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---unshallow)
2. when running locally, we have full clones so unshallow fails